semgrep-core/src/core/Rule.ml

```diff
@@ -209,6 +209,7 @@ type extract_spec = {
   dst_lang : Xlang.t;
   (* e.g., $...BODY, $CMD *)
   extract : MV.mvar;
+  json : bool;
```

Comment:
I think there are a few possible scenario where the extracted code is not in final form and need to be further
transformed. 'json:' is actually not a good name because in this case it's actually a json string array.
Maybe we could have a field like 'transform: concat_json_string_array' where we could have a list of possible
choices for this new transform: field. we could also have 'transform: unquote_string'.
Ideally the extract rules would have done the job, and reduce: concat could do things, but maybe sometimes it's hard and an additional transform: field could help.

Reply:
Yeah, I thought something like that, based on a conversation that we had in Slack but, since someone mentioned that this feature may be redesigned by the end of the year, I didn't think it was critical to do something more complex, since it could be completely changed in some months.

Reply:
still, 'json:' is currently a bad name, because it should really be concat_json_string_array: true, at which point
doing an enum with transform: concat_json_string_array | unqote_string seems better.

Reply:
This has been reimplemented as suggested.